clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -101,21 +103,80 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
       String classicLinkVpcId = null
       List<String> classicLinkVpcSecurityGroups = null
       if (!subnetType) {
-        classicLinkVpcId = description.classicLinkVpcId
-        if (!classicLinkVpcId) {
-          def result = amazonEC2.describeVpcClassicLink()
-          def classicLinkVpc = result.vpcs.find { it.classicLinkEnabled }
-          if (classicLinkVpc) {
-            classicLinkVpcId = classicLinkVpc.vpcId
-          }
-        }
+        def result = amazonEC2.describeVpcClassicLink()
+        classicLinkVpcId = result.vpcs.find { it.classicLinkEnabled }?.vpcId
         if (classicLinkVpcId) {
-          classicLinkVpcSecurityGroups = description.classicLinkVpcSecurityGroups
-          if (!classicLinkVpcSecurityGroups) {
-            if (deployDefaults.classicLinkSecurityGroupName) {
-              classicLinkVpcSecurityGroups = [deployDefaults.classicLinkSecurityGroupName]
+          Set<String> classicLinkGroupNames = []
+          classicLinkGroupNames.addAll(description.classicLinkVpcSecurityGroups ?: [])
+          if (deployDefaults.classicLinkSecurityGroupName) {
+            classicLinkGroupNames.addAll(deployDefaults.classicLinkSecurityGroupName)
+          }
+
+          // if we have provided groups and a vpcId, resolve them back to names to handle the case of cloning
+          // from a Server Group in a different region
+          if (description.classicLinkVpcId && description.classicLinkVpcSecurityGroups) {
+            def groupIds = classicLinkGroupNames.findAll { it.matches(~/sg-[0-9a-f]+/) } ?: []
+            classicLinkGroupNames.removeAll(groupIds)
+            if (groupIds) {
+              def describeSG = new DescribeSecurityGroupsRequest().withGroupIds(groupIds)
+              def provider = sourceRegionScopedProvider ?: regionScopedProvider
+              def resolvedNames = provider.amazonEC2.describeSecurityGroups(describeSG).securityGroups.findResults {
+                if (it.vpcId == classicLinkVpcId && groupIds.contains(it.groupId)) {
```

Comment:
I wasn't sure if the DescribeSecurityGroups would fail on an unknown groupId or just return the subset that were found. Because they potentially come in from an API request I figured validating they were legit during the lookup made sense.